App/src/main/res/layout/activity_settings.xml

```diff
@@ -1,105 +1,74 @@
-        android:textColor="@android:color/black"
-        android:textSize="12sp"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/user_name"
         android:layout_centerHorizontal="true"
-        android:text="Your Lokki ID is user@email.com"
         android:padding="0dip"
-        android:layout_below="@id/user_name"
-        android:layout_height="wrap_content"
-        android:layout_width="wrap_content" />
+        android:text=""
+        android:textColor="@android:color/black"
+        android:textSize="12sp" />
 
     <TextView
         android:id="@+id/your_friends"
-        android:textColor="@android:color/darker_gray"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/lokki_id_text"
         android:layout_centerHorizontal="true"
-        android:text="@string/your_friends"
         android:padding="0dip"
+        android:text="@string/your_friends"
+        android:textColor="@android:color/darker_gray"
         android:textSize="12sp"
-        android:textAlignment="center"
-        android:layout_below="@id/lokki_id_text"
-        android:layout_height="wrap_content"
-        android:layout_width="240dp" />
-
-    <TextView
-        android:id="@+id/visibility"
-        android:textColor="@android:color/black"
-        android:text="@string/my_visibility"
-        android:textStyle="italic"
-        android:padding="5dip"
-        android:textSize="18sp"
-        android:textAlignment="center"
-        android:layout_height="wrap_content"
-        android:layout_width="wrap_content"
-        android:layout_marginTop="20dp"
-        android:layout_below="@+id/your_friends" />
-
-    <Spinner
-        android:layout_width="fill_parent"
-        android:layout_height="wrap_content"
-        android:id="@+id/spinner_visibility"
-        android:layout_below="@+id/visibility"
-        android:paddingLeft="8dp"
-        android:paddingStart="8dp"/>
-
-    <TextView
-        android:id="@+id/map_type"
-        android:textColor="@android:color/black"
-        android:textStyle="italic"
-        android:text="@string/map_mode"
-        android:padding="5dip"
-        android:textSize="18sp"
-        android:layout_marginTop="10dp"
-        android:layout_height="wrap_content"
-        android:layout_width="wrap_content"
-        android:layout_below="@+id/spinner_visibility" />
+        android:gravity="center_horizontal" />
+    </RelativeLayout>
 
-    <Spinner
+    <fragment
         android:layout_width="fill_parent"
         android:layout_height="wrap_content"
-        android:id="@+id/spinner_map"
-        android:layout_below="@+id/map_type"
-        android:paddingLeft="8dp"
-        android:paddingStart="8dp" />
+        android:layout_margin="0dp"
+        android:name="cc.softwarefactory.lokki.android.PreferencesFragment"
+        android:id="@+id/fragment_settings"/>
```

Comment:
Having a fragment inside a fragment is probably not very stable. Should change the `SettingsFragment` into its own `SettingsActivity`.